pkg/question/question.go

```diff
@@ -1229,127 +1236,173 @@ func AskInstanceIds(h *ec2helper.EC2Helper, addedInstanceIds []string) (*string,
 	question := fmt.Sprintf("Are you sure you want to terminate %d instance(s): %s ", len(instanceIds), instanceIds)
+	answer, err := askYesNoQuestion(question, false)
 
-	answer := AskQuestion(&AskQuestionInput{
-		QuestionString: question,
-		DefaultOption:  aws.String(cli.ResponseNo),
-		OptionsString:  &optionsText,
-		StringOptions:  stringOptions,
-	})
+	if err != nil {
+		return "", err
+	}
 
-	return answer
+	return answer, nil
 }
 
-func AskCapacityType(instanceType string, defaultCapacityType string) string {
-	ec2Pricing := ec2pricing.New(session.New())
+func AskCapacityType(instanceType string, region string, defaultCapacityType string) (string, error) {
+	ec2Pricing := ec2pricing.New(session.New().Copy(aws.NewConfig().WithRegion(region)))
 	onDemandPrice, err := ec2Pricing.GetOnDemandInstanceTypeCost(instanceType)
-	formattedOnDemandPrice := ""
+	formattedOnDemandPrice := "N/A"
 	if err == nil {
+		fmt.Printf("onDemandPrice: %v\n", onDemandPrice)
 		onDemandPrice = math.Round(onDemandPrice*10000) / 10000
-		formattedOnDemandPrice = fmt.Sprintf("($%s/hr)", strconv.FormatFloat(onDemandPrice, 'f', -1, 64))
+		formattedOnDemandPrice = fmt.Sprintf("$%s/hr", strconv.FormatFloat(onDemandPrice, 'f', -1, 64))
 	}
 
 	spotPrice, err := ec2Pricing.GetSpotInstanceTypeNDayAvgCost(instanceType, []string{}, 1)
-	formattedSpotPrice := ""
+	formattedSpotPrice := "N/A"
 	if err == nil {
 		spotPrice = math.Round(spotPrice*10000) / 10000
-		formattedSpotPrice = fmt.Sprintf("($%s/hr)", strconv.FormatFloat(spotPrice, 'f', -1, 64))
+		formattedSpotPrice = fmt.Sprintf("$%s/hr", strconv.FormatFloat(spotPrice, 'f', -1, 64))
 	}
 
 	question := fmt.Sprintf("Select capacity type. Spot instances are available at up to a 90%% discount compared to On-Demand instances,\n" +
 		"but they may get interrupted by EC2 with a 2-minute warning")
 
-	optionsText := fmt.Sprintf("1. On-Demand%s\n2. Spot%s\n", formattedOnDemandPrice,
-		formattedSpotPrice)
 	indexedOptions := []string{DefaultCapacityTypeText.OnDemand, DefaultCapacityTypeText.Spot}
 	defaultOption := DefaultCapacityTypeText.OnDemand
 	if slices.Contains(indexedOptions, defaultCapacityType) {
 		defaultOption = defaultCapacityType
 	}
 
-	answer := AskQuestion(&AskQuestionInput{
+	data := [][]string{{DefaultCapacityTypeText.OnDemand, formattedOnDemandPrice}, {DefaultCapacityTypeText.Spot, formattedSpotPrice}}
+
+	headers := []string{"Capacity Type", "Price"}
+
+	model := &questionModel.SingleSelectList{}
+	err = questionModel.AskQuestion(model, &questionModel.QuestionInput{
 		QuestionString: question,
-		DefaultOption:  &defaultOption,
-		OptionsString:  &optionsText,
+		DefaultOption:  defaultOption,
 		IndexedOptions: indexedOptions,
+		OptionData:     data,
+		HeaderStrings:  headers,
+	})
+
+	if err != nil {
+		return "", err
+	}
+
+	return model.GetChoice(), nil
+}
+
+func askYesNoQuestion(question string, defaultToYes bool) (string, error) {
```

Comment:
This might deserve a standalone file in the `questionModel` folder

Reply:
How so? Its just a function to ask the same question instead of rewriting it in multiple places?

Reply:
Its not creating a new question type

Reply:
It's a judgment call either way--go ahead and leave it here for now.

As you say, under the hood it's not a new question type because it wraps singleSelect. But logically it is a distinct kind of question that is used more than once, so I could make an argument that it deserves to be a first-class question type on its own, even though it's just a special case of singleSelect.